packages/babel-traverse/src/scope/index.js

```diff
@@ -514,6 +515,8 @@ export default class Scope {
           if (local.identifier === id) continue;
 
           this.checkBlockScopedCollisions(local, kind, name, id);
+          // if no duplicates were found, set existing binding to null
+          local = null
```

Comment:
this shouldn't be needed because `local` is block scoped. And the next loop will get a fresh binding